src/pydisagg/ihme/splitter/sex_splitter.py

```diff
@@ -382,19 +381,14 @@ def split(
         # Step 2: Handle rows that need to be split (where `sex_id` is not `sex_m` or `sex_f`)
         split_data = data[~mask_sex_m_or_f].copy()
 
-        # Calculate input patterns and splitting logic
+        input_patterns = np.vstack(
+            [np.ones(len(split_data)), split_data[self.pattern.val].values]
+        ).T
```

Comment:
I think for categorical splitter, we don't need this, user should provide the pattern rather than the ratio.